daemon/daemon_unix.go

```diff
@@ -836,42 +836,53 @@ func configureKernelSecuritySupport(config *config.Config, driverName string) er
 	return nil
 }
 
-func (daemon *Daemon) initNetworkController(activeSandboxes map[string]interface{}) (libnetwork.NetworkController, error) {
+// initNetworkController initializes the libnetwork controller and configures
+// network settings. I there's active sandboxes, configuration changes will not
+// take effect.
+func (daemon *Daemon) initNetworkController(activeSandboxes map[string]interface{}) error {
 	netOptions, err := daemon.networkOptions(daemon.PluginStore, activeSandboxes)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	controller, err := libnetwork.New(netOptions...)
+	daemon.netController, err = libnetwork.New(netOptions...)
 	if err != nil {
-		return nil, fmt.Errorf("error obtaining controller instance: %v", err)
+		return fmt.Errorf("error obtaining controller instance: %v", err)
 	}
 
-	conf := daemon.configStore
 	if len(activeSandboxes) > 0 {
-		logrus.Info("There are old running containers, the network config will not take affect")
-		setHostGatewayIP(conf, controller)
-		return controller, nil
+		logrus.Info("there are running containers, updated network configuration will not take affect")
+	} else {
+		err := configureNetworking(daemon.netController, daemon.configStore)
```

Comment:
Same nit: we can have `if err := ...; err != nil {` here

Reply:
done 👍